mesonbuild/build.py

```diff
@@ -377,11 +377,24 @@ class IncludeDirs(HoldableObject):
     # Interpreter has validated that all given directories
     # actually exist.
     extra_build_dirs: T.List[str] = field(default_factory=list)
+    build_relative: bool = field(default=True)
+    source_relative: bool = field(default=True)
 
     def __repr__(self) -> str:
         r = '<{} {}/{}>'
         return r.format(self.__class__.__name__, self.curdir, self.incdirs)
 
+    def __post_init__(self):
```

Comment:
I'm not sure this validation step is really necessary. User arguments are already checked in the interpreter.

Reply:
I believe this overlaps with what I mentioned [above](https://github.com/mesonbuild/meson/pull/11984#discussion_r1266562204):   There are at least 5 different places that create IncludeDirs instances:  Some come through the interpreter (which we expect to do user-facing validation) but the other instances are from meson internals, which, along with any new future additional internal use, can benefit from the assert catching future internal/meson-programmer misuse.